// babel-plugin-module-resolve/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var resolve_1 = require("./resolve");
var normalize_options_1 = __importDefault(require("./normalize-options"));
var transform_1 = require("./transform");
var plugin = function (_a) {
    var t = _a.types;
    return {
        name: 'module-resolve',
        pre: function () {
            this.cwd = this.file.opts.root;
            this.resolver = resolve_1.createResolver(normalize_options_1["default"].call(this));
            this.visitedNode = new Set();
            this.t = t;
        },
        post: function () {
            this.visitedNode.clear();
        },
        visitor: {
            ImportDeclaration: transform_1.transformImport,
            ExportNamedDeclaration: transform_1.transformImport,
            ExportAllDeclaration: transform_1.transformImport,
            CallExpression: transform_1.transformCall
        }
    };
};
exports["default"] = plugin;
